library/src/scripts/components/Heading.tsx

```diff
@@ -31,6 +32,8 @@ export default class Heading extends React.Component<IHeadingProps> {
         depth: 2,
     };
 
+    private renderAsDepth = this.props.renderAsDepth ? this.props.renderAsDepth : this.props.depth;
```

Comment:
This assignment works and in most cases would be fine, unless one of these 2 props were changed.

In this instance a computer getter is more flexible, as it will be re-computed on subsequent updates.

```suggestion
    private get renderAsDepth(): number {
        return this.props.renderAsDepth ? this.props.renderAsDepth : this.props.depth!;
    }
```